api/app/hfi/hfi_request.py

```diff
@@ -0,0 +1,37 @@
+""" HFI request logic """
+
+from itertools import groupby
+import operator
+from typing import List
+from app.db.models.hfi_calc import PlanningWeatherStation
+from app.schemas.hfi_calc import HFIResultRequest, StationInfo
+
+
+def update_result_request(result_request: HFIResultRequest,
+                          latest_stations: List[PlanningWeatherStation]):
+    """ Immutable copy of the result request with the latest planning weather stations """
```

Comment:
Maybe just lack of sleep, but - it's not immutable is it?
```suggestion
    """Create copy of the result request with the latest planning weather stations """
```

Reply:
Input is immutable, so yeah, not the best comment